isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+#include "Isis.h"
+
+#include "Application.h"
+#include "Cube.h"
+#include "CubeAttribute.h"
+#include "FileName.h"
+#include "Portal.h"
+#include "ProcessByLine.h"
+#include "Pvl.h"
+#include "SpecialPixel.h"
+#include "UserInterface.h"
+
+#include <QString>
+
+using namespace std;
+using namespace Isis;
+
+// Taken from ProcessMosaic
+const int FLOAT_MIN = -16777215;
+const int FLOAT_MAX = 16777216;
+
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand);
+void createMosaicCube(UserInterface ui, QString bands);
+void createTrackCube(UserInterface ui, QString trackingBand);
+void copyPixels(Buffer &in, Buffer &out);
+void copyTrackPixels(Buffer &in, Buffer &out);
+
+void IsisMain() {
+  UserInterface &ui = Application::GetUserInterface();
+  QString copyBands;
+  QString trackBand;
+  findTrackBand(ui, copyBands, trackBand);
+  createMosaicCube(ui, copyBands);
+  createTrackCube(ui, trackBand);
+}
+
+
+/**
+ * Finds the index of the tracking band in the BandBin group and stores the index
+ * in trackBand. The other bands' indices are stored in copyBands.
+ * We store the indices in QStrings because we use them as cube attributes when
+ * we process the mosaic and tracking cubes.
+ *
+ * @param ui        The user interface
+ * @param copyBands Indices of the bands to copy over to the mosaic cube
+ * @param trackBand Index of the tracking band
+ */
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand) {
```

Comment:
You should directly pass the cube filename instead of the user interface. You could also just pass a pointer to the cube. Passing the user interface is opaque, it's not clear what it's going to be used for. This way, if we ever move this out of this application code we can just copy and past it over.